eiffel-syntax-and-usage/the-links-object.md

```diff
@@ -16,15 +16,31 @@
 --->
 
 # The Links Object
-The __links__ object is an array of trace links to other Eiffel events. These trace links by definition always reference backwards in time – it is only possible to reference an event that has already occurred. Each trace link is a tuple consisting of a type and a UUID corresponding to the __meta.id__ of the target event, on String format.
+The __links__ object is an array of trace links to other Eiffel events. These trace links by definition always reference backwards in time – it is only possible to reference an event that has already occurred. Each trace link is a tuple consisting of a type, a UUID corresponding to the __meta.id__ of the target event, on String format, and optionally the id of the [domain](glossary.md#domain) where the target event was published (i.e. its __meta.source.domainId__ member). The absence of a domain id means that the target event is in the same domain as the current event.
```

Comment:
Not really related to your change, but each link is an object right? Not a tuple.

Reply:
Yes, from a JSON perspective it's clearly an object. The original sentence probably tries to use a more general (or academic?) language, but that stands in contrast to the start of the bullet's

 > The links object is an array of trace links to other Eiffel events

which explicitly speaks of an array. For consistency maybe the bullet should start

> The links key contains a set of trace links to other Eiffel events

or we could use "object" instead of "tuple". I have no particular opinion.

Reply:
I was struggling to figure out where `on String format` should fit. Could something like this be more readable?

```suggestion
The __links__ object is an array of trace links to other Eiffel events. These trace links by definition always reference backwards in time – it is only possible to reference an event that has already occurred. Each trace link is a tuple consisting of:
- a type
- a UUID corresponding to the __meta.id__ of the target event, on String format, 
- optionally the id of the [domain](glossary.md#domain) where the target event was published (i.e. its __meta.source.domainId__ member). The absence of a domain id means that the target event is in the same domain as the current event.
```

Reply:
I could also go with object instead of tuple.

Reply:
Yeah, breaking it up into a bullet list looks better. Done.